x.toml

```diff
@@ -156,3 +156,26 @@ members = [
     "secure/key-manager",
     "storage/backup/backup-cli",
 ]
+
+# ---
+# Determinator rules
+# ---
+
+# CI-related files. TODO: maybe have separate rules for local and CI?
+[[determinator.path-rule]]
+globs = [".github/**/*", ".circleci/**/*", "codecov.yml"]
+mark-changed = "all"
```

Comment:
I personally think (based on what I implement with maven years ago doing something similar) that all files not explicitly excluded from the root of the project should be treated as changes to the root cargo project and result in a full rebuild.   Basically arguing the general assumption that any file under a Cargo.toml is part of the project unless there is another Cargo.toml between it and the root...   closest wins.   Everything after that is an exclusion.
This would capture changes to text files that are referenced in tests for example.    This rule would be redundant in that case.

Reply:
Not exactly true in the case of crates that aren't listed as packages from the virtual cargo toml.   I think we should remove devtools as default targets so the "fall back in to" being part of the root virtual crate.

Reply:
Hmm, I've gone back and forth on whether the default for a file that can't be matched to a closer Cargo.toml should be skip or full-rebuild. The arguments for skip are:
* We don't try and represent the virtual package at the root at all -- we only care about the actual packages with real code.
* If the root `Cargo.toml` or `Cargo.lock` changes we *do not* do a full rebuild -- and this isn't some sort of special optimization, it falls naturally out of the design.
* In my experience, overbuilding is harder to notice than underbuilding unless we add some logging and metrics which we don't have at the moment -- missing a bug because of underbuilding will be caught quickly.

The arguments for full-rebuild are:
* It's less likely to lead to bugs than skip.
* Maybe it matches people's intuitions better? Not sure about this one.

Some other options are:
* Add a flag to the determinator config controlling the default.
* Print clear warnings for files that were ignored to logs.

Either way, it makes sense that some files should probably be added to the default set of rules in the determinator upstream, like `rust-toolchain` etc. I'll do that upstream.

>  Basically arguing the general assumption that any file under a Cargo.toml is part of the project unless there is another Cargo.toml between it and the root... closest wins

To be clear, this is already how the determinator works, except we don't have a notion of "part of the project" in the determinator at all.

Reply:
After discussion with @rexhoffman, switched the default over to full-rebuild rather than ignore. Also added a linter to ensure that every file in the libra workspace is matched by a rule and/or a package.